lib/fs.js

```diff
@@ -1794,10 +1800,13 @@ function WriteStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
-
-  options = Object.create(options);
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    internalUtil.printDeprecationMessage('Passing anything but an object or \
+a string as the second argument to WriteStream is deprecated.');
+    options = options || {};
+  } else
```

Comment:
did you run `make jslint` ? eslint is sometimes buggy when there is a mix of `else` statements with or without brackets.